classes/abstracts/ActionScheduler.php

```diff
@@ -95,7 +95,7 @@ public static function autoload( $class ) {
 
 			switch ( $type ) {
 				case 'WPCLI':
-					$dir = $classes_dir . 'WP_CLI' . $d;
+					$dir = $classes_dir . 'WPCLI' . $d;
```

Comment:
any specific reason as to why this folder name is not changed to `WP-CLI` instead like in other cases?

Reply:
> Is it feasible to add files for old classes with a require_once realpath(..) statement and a deprecation notice?

Any plugin requiring an AS class file directly risks installs where AS is broken, logging errors, and/or fatal errors on the site. There were several plugins that had to address this for AS 3.0. The only file that should be accessed outside of AS is `action-scheduler/action-scheduler.php`.

> any specific reason as to why this folder name is not changed to WP-CLI instead like in other cases?

The `_` -> `-` string replacement is done on the filename not the folder. I would have removed the `_` from the filenames instead of replacing except it made some of the class file names difficult to visually parse. Contributors with easily recognize `WPCLI` as it is often referenced in online content without the `_`.

Reply:
I see, but we need to mark it as such when we update the version in the WC core.

Reply:
I am going to test a bit more before approving, but looks good so far.